src/scripts/main.js

```diff
@@ -1,3 +1,113 @@
 'use strict';
 
-// write code here
+const buttons = Array.from(document.querySelectorAll('button'));
+const tableBody = document.querySelector('.field tbody');
+const trElements = document.querySelector('tr');
+
+buttons.forEach((button) => {
+  button.onclick = function (events) {
+    const target = events.target.className;
+
+    switch (target) {
+      case 'append-row button':
+        appendRow();
+        break;
+
+      case 'remove-row button':
+        removeRow();
+        break;
+
+      case 'append-column button':
+        appendColumn();
+        break;
+
+      case 'remove-column button':
+        removeColumn();
+        break;
+    }
+  };
+});
+
+function appendRow() {
+  const rowCount = tableBody.children.length;
+  const removeButton = document.querySelector('.remove-row');
+
+  if (removeButton.hasAttribute('disabled')) {
+    removeButton.removeAttribute('disabled');
+  }
+
+  if (rowCount < 10) {
+    const tr = document.createElement('tr');
+
+    for (let i = 0; i < trElements.children.length; i++) {
```

Comment:
Using `trElements.children.length` assumes `trElements` is a collection of `tr` elements. However, it is currently a single `tr` element. This will cause issues when trying to append columns. Consider using a collection of `tr` elements.